iceoryx_hoofs/include/iceoryx_hoofs/posix_wrapper/command_line.hpp

```diff
@@ -0,0 +1,173 @@
+// Copyright (c) 2022 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+#ifndef IOX_HOOFS_POSIX_WRAPPER_COMMAND_LINE_HPP
+#define IOX_HOOFS_POSIX_WRAPPER_COMMAND_LINE_HPP
+
+#include "iceoryx_hoofs/cxx/convert.hpp"
+#include "iceoryx_hoofs/cxx/function.hpp"
+#include "iceoryx_hoofs/cxx/type_traits.hpp"
+#include "iceoryx_hoofs/cxx/vector.hpp"
+#include "iceoryx_hoofs/internal/posix_wrapper/command_line_parser.hpp"
+
+namespace iox
+{
+namespace posix
+{
+namespace internal
+{
+using cmdEntries_t = cxx::vector<CommandLineParser::Entry, CommandLineOptions::MAX_NUMBER_OF_ARGUMENTS>;
+using cmdAssignments_t =
```

Comment:
@elBoberido This is correct. We have also no clang-tidy rule for this. I could add a rule for:
TypeAliasCase, TypeAliasPrefix, TypeAliasSuffix in `readability-identifier-naming` section of the clang tidy file.

I have no passion here but I like the lower camel case and the `_t` suffix. The google guidelines seem to have the same style like for classes upper camel case.

I suggest we decide for one option here, I add this to the central clang-tidy file and when someone does not agree with this, this can be changed quickly. Afterall the clang-tidy file should be viewed as experimental and is still undergoing change and is only checked for new files.